lib/ansible/modules/cloud/google/gcp_spanner_instance.py

```diff
@@ -0,0 +1,345 @@
+        returned: success
+        type: str
+    config:
+        description:
+            - A reference to the instance configuration.
+        returned: success
+        type: str
+    display_name:
+        description:
+            - The descriptive name for this instance as it appears in UIs. Must be unique per
+              project and between 4 and 30 characters in length.
+        returned: success
+        type: str
+    node_count:
+        description:
+            - The number of nodes allocated to this instance.
+        returned: success
+        type: int
+    labels:
+        description:
+            - Cloud Labels are a flexible and lightweight mechanism for organizing cloud resources
+              into groups that reflect a customer's organizational needs and deployment strategies.
+              Cloud Labels can be used to filter collections of resources. They can be used to
+              control how resource metrics are aggregated. And they can be used as arguments to
+              policy management rules (e.g. route, firewall, load balancing, etc.).
+            - 'Label keys must be between 1 and 63 characters long and must conform to the following
+              regular expression: `[a-z]([-a-z0-9]*[a-z0-9])?`.'
+            - Label values must be between 0 and 63 characters long and must conform to the regular
+              expression `([a-z]([-a-z0-9]*[a-z0-9])?)?`.
+            - No more than 64 labels can be associated with a given resource.
+            - See U(https://goo.gl/xmQnxf) for more information on and examples of labels.
+            - 'If you plan to use labels in your own code, please note that additional characters
+              may be allowed in the future. And so you are advised to use an internal label representation,
+              such as JSON, which doesn''t rely upon specific characters being disallowed. For
+              example, representing labels as the string: name + "_" + value would prove problematic
+              if we were to allow "_" in a future release.'
+            - 'An object containing a list of "key": value pairs.'
+            - 'Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.'
+        returned: success
+        type: dict
+'''
+
+################################################################################
+# Imports
+################################################################################
+
+from ansible.module_utils.gcp_utils import navigate_hash, GcpSession, GcpModule, GcpRequest, replace_resource_dict
+import json
+
+################################################################################
+# Main
+################################################################################
+
+
+def main():
+    """Main function"""
+
+    module = GcpModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            name=dict(type='str'),
```

Comment:
Wouldn't a display name also be required? Otherwise how would the API distinguish which instance is being referred to?

Reply:
Look down a couple lines :)